exporter/exporterhelper/internal/persistent_storage.go

```diff
@@ -154,9 +156,22 @@ func (pcs *persistentContiguousStorage) Capacity() int {
 	return int(pcs.capacity)
 }
 
-func (pcs *persistentContiguousStorage) stop(ctx context.Context) error {
-	pcs.logger.Debug("Stopping persistentContiguousStorage")
-	return pcs.client.Close(ctx)
+func (pcs *persistentContiguousStorage) Shutdown(ctx context.Context) error {
+	close(pcs.stopChan)
+	// Hold the lock only for `refClient`.
+	pcs.mu.Lock()
+	defer pcs.mu.Unlock()
```

Comment:
Why not move the locking to unrefClient?

Reply:
Because I want to execute this and the `itemDispatchingFinish` while holding the lock, so in the `onProcessingFinishedFunc` I don't have to release the lock and re-acquire.

Reply:
I see 👍